src/GafferScene/Wireframe.cpp

```diff
@@ -0,0 +1,256 @@
+
+#include "GafferScene/Wireframe.h"
+
+#include "Gaffer/StringPlug.h"
+
+#include "IECoreScene/MeshPrimitive.h"
+#include "IECoreScene/CurvesPrimitive.h"
+
+#include "IECore/DataAlgo.h"
+
+#include "boost/functional/hash.hpp"
+
+#include <unordered_set>
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+//////////////////////////////////////////////////////////////////////////
+// Internal utilities
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+struct MakeWireframe
+{
+
+	CurvesPrimitivePtr operator() ( const V2fVectorData *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		return makeWireframe( positions->baseReadable(), 2, verticesPerFace, vertexIds );
+	}
+
+	CurvesPrimitivePtr operator() ( const V3fVectorData *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		return makeWireframe( positions->baseReadable(), 3, verticesPerFace, vertexIds );
+	}
+
+	CurvesPrimitivePtr operator() ( const Data *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		throw IECore::Exception( boost::str(
+			boost::format( "Position has unsupported type \"%1%\"" ) % positions->typeName()
+		) );
+	}
+
+	private :
+
+		V3f position( const float *positions, int dimensions, int index )
+		{
+			V3f result( 0.0f );
+			for( int i = 0; i < dimensions; ++i )
+			{
+				result[i] = positions[index*dimensions+i];
+			}
+			return result;
+		}
+
+		CurvesPrimitivePtr makeWireframe( const float *positions, int dimensions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+		{
+			IECore::V3fVectorDataPtr pData = new V3fVectorData;
+			pData->setInterpretation( GeometricData::Point );
+			vector<V3f> &p = pData->writable();
+
+			using Edge = std::pair<int, int>;
+			using EdgeSet = unordered_set<Edge, boost::hash<Edge>>;
+			EdgeSet edgesVisited;
+
+			int vertexIdsIndex = 0;
+			for( int numVertices : verticesPerFace )
+			{
+				for( int i = 0; i < numVertices; ++i )
+				{
+					int index0 = vertexIds[vertexIdsIndex + i];
+					int index1 = vertexIds[vertexIdsIndex + (i + 1) % numVertices];
+					Edge edge( min( index0, index1 ), max( index0, index1 ) );
+					if( edgesVisited.insert( edge ).second )
+					{
+						p.push_back( position( positions, dimensions, index0 ) );
+						p.push_back( position( positions, dimensions, index1 ) );
+					}
+				}
+				vertexIdsIndex += numVertices;
+			}
+
+			IECore::IntVectorDataPtr vertsPerCurveData = new IntVectorData;
+			vertsPerCurveData->writable().resize( p.size() / 2, 2 );
+
+			CurvesPrimitivePtr result = new CurvesPrimitive( vertsPerCurveData );
+			result->variables["P"] = PrimitiveVariable( PrimitiveVariable::Vertex, pData );
+			return result;
+		}
+
+};
+
+/// \todo Perhaps this could go in IECoreScene::MeshAlgo
```

Comment:
I think this should be moved and the edge data structure perhaps?

Reply:
I'll move the algo once we've merged and I've done some further practical testing - I'm not of a mind to do yet another dependencies release prior to 0.53. I don't think the edge data structure belongs in any public API - it's not generally applicable enough.